onnx/reference/ops/op_attention.py

```diff
@@ -24,6 +24,19 @@ def _softcap(X, softcap):
     return X
 
 
+def apply_causal(mask, inplace=False):
+    q_sequence_length, total_sequence_length = mask.shape[-2:]
+    past_sequence_length = total_sequence_length - q_sequence_length
+    triu = np.triu(
+        np.ones((q_sequence_length, q_sequence_length), dtype=mask.dtype), k=1
+    )
+    triu[triu == 1] = -np.inf
```

Comment:
When mask is bool array, you are trying to write float into it.

Reply:
fixed